claim-check-pattern/ReadME.md

```diff
@@ -0,0 +1,131 @@
+
+
+
+In Plain words
+> Split a large message into a claim check and a payload. Send the claim check to the messaging platform and store the payload to an external service. This pattern allows large messages to be processed while protecting the message bus and the client from being overwhelmed or slowed down. This pattern also helps to reduce costs, as storage is usually cheaper than resource units used by the messaging platform.([ref](https://docs.microsoft.com/en-us/azure/architecture/patterns/claim-check))
+
+## Architecture Diagram
+![alt text](./etc/Claim-Check-Pattern.png "Claim Check Pattern")
+
+## Applicability
+Use the Claim Check Pattern when
+- processing data is huge and don't want to consume bandwidth to transfer data through internet.
+- secure your data transfer by storing in common persistent storage.
+- you have cloud platform - Azure Functions or AWS Lambda, Azure EventGrid or AWS Event Bridge, Azure Blob Storage or AWS S3 Bucket.
+- each service must be independent. It means it will perform its operations and drop data to storage.
+- Pattern follows Publish-Subscribe Model.
+
+## Consequences
+- This pattern is stateless pattern. Any compute API will not store any data.
+- You must have persistent storage and reliable messaging platform.
+
+## Class Diagrams
+
+### usage-detail-sender (Producer)
+![alt text](./etc/Usage-Detail-Sender_Class_Diagram.urm.png "Adapter class diagram")
+
+### usage-cost-processor (Consumer)
+![alt text](./etc/Usage-Cost-Processor_Class_Diagram.urm.png "Adapter class diagram")
+
+## Tutorials
+
+### Workflow
+We are building call cost calculator system. Producer class will generate call usage details and send it to kafka topic after each 5 secs. Consumer class will read this data from kafka topic and calculate cost. It then store its result in storage. First, usage-detail-sender creates a message, sends message header to Kafka topic usage-detail, and drops an entire message to the local file system. Kafka then sent this message header to the usage-cost-processor service. It will read the entire message with the help of the header, perform its operation, and drop the result to the local system.
+
+### Setup
+- Make sure your system is windows.
+- Run everything as administrator.
+- Install Java JDK 11 and setup environmental variables.
+- Install git.
+- Download Apache kafka binaries and extract.
+- Install Spring Tool Suite 4 LTS
+
+### Command for Apache Kafka
+
+I am using git bash for the following commands to execute.
```

Comment:
Leave out the first person and just say that Git Bash should be used to execute the following commands

Reply:
@iluwatar Removed.